pkg/config/xds/xds.go

```diff
@@ -78,15 +76,15 @@ func StructToMessage(pbst *structpb.Struct, out proto.Message, strict bool) erro
 		return errors.New("nil struct")
 	}
 
-	buf := &bytes.Buffer{}
-	if err := (&jsonpb.Marshaler{OrigName: true}).Marshal(buf, pbst); err != nil {
```

Comment:
does missing origname matter?

Reply:
Tests pass so maybe not....

Reply:
`MarshalProtoNames` has OrigName true.
It's your impl :)